solution/flask_apps/services/resources/abstract_resource.py

```diff
@@ -0,0 +1,80 @@
+import abc
+import re
+
+from .gspread_service import GspreadService
+
+import pandas as pd
+
+gsheets_dict = {
+    'admissions': '1zG8i_IqstrIenDAmQTtgdZ7lILKHY5x_v8jHKgkyDcs',
+    'enrollments': '18tvRHT4qy15cSnC9tkrs5I1GkDfqzjC3LFaaUuy6jPg',
+    'outcomes': '1yN-9oTIuF69w57oTzvQpEbfd3fmmRKnhhNvBUKJXdnc'
+}
+
+# Helper function
+def sheet_to_df(sheet):
+    data = sheet.get_all_values()
+    df = pd.DataFrame(data)
+    df.columns = df.iloc[0]
+    df = df.iloc[1:]
+    return df
+
+class AbstractResource(abc.ABC):
+
+    @abc.abstractmethod
+    def fetch(self):
+        pass
+    
+    @abc.abstractmethod
+    def process(self):
+        pass
+    
+    @abc.abstractmethod
+    def store(self):
+        pass
+
+class AbstractSpreadSheetResource(AbstractResource):
+    def __init__(self):
+        self.sheet_id = None  # google sheet id
+        self.parsers = {} # sheet with parser function
+        self.gsp_service = GspreadService()
+        self.raw_data = {} # sheet name with raw data
+        self.parsed_data = {} # sheet name with cleaned data
+
+    def fetch(self):
+        if self.sheet_id:
+            self.raw_data = self.fetch_from_workbook(self.sheet_id)
+        else:
+            print('No Sheet_id given')
```

Comment:
it would be nice if you could fetch by sheet name or just a range from a sheet too, sometimes its not feasible to fetch an entire spreadsheet when you only need information from a single sheet.